### PR TITLE
refactor getGroup/getGroups

### PR DESCRIPTION
**Pull request by soltanireza65**
instead of calling groupManager->get($gid);
now we are instantiating a new Group object directly in `handleUpdateGroup`
```php
    private function getGroupObject(string $groupId): Group {
        return new Group($groupId, [$this->groupBackend], $this->userManager, $this->eventDispatcher, $this->groupManager, $groupId);
    }
```
```php
private function handleUpdateGroup(string $groupId, $obj) {
        $group = $this->getGroupObject($groupId);
		...
}
```
plus some refactoring to cleanup getGroup/s

issue link: https://github.com/SURFnet/rd-sram-integration/issues/197

### CODE REVIEW

File: federatedgroups/lib/Controller/ScimController.php [hunk 1]
```diff
@@ -139,21 +147,24 @@ public function createGroup($id, $members) {
...
+        // $group = new Group($id, [$this->groupBackend], $this->userManager, $this->eventDispatcher, $this->groupManager, $id);
+
 
         // expect group to already exist
         // we are probably receiving this create due to
         // https://github.com/SURFnet/rd-sram-integration/commit/38c6289fd85a92b7fce5d4fbc9ea3170c5eed5d5
         try {
             $this->handleUpdateGroup($id, $body);
+            // $groupData = $this->handleGetGroupData($id);
+            // return new JSONResponse(['status'  => 'success', 'message' => null, 'data' => $groupData], Http::STATUS_CREATED);
```
Comment (thepeak99) [inline federatedgroups/lib/Controller/ScimController.php:167]: why do you leave this commented code?

**Comment by thepeak99** [comment]
Hi! why are you creating a new object every time? Maybe there was a discussion that I missed :cry:

File: federatedgroups/lib/Controller/ScimController.php (hunk 1 above)
Comment (soltanireza65) [reply federatedgroups/lib/Controller/ScimController.php]: removed them

**Comment by soltanireza65** [comment]
@thepeak99 
can you take a look at the issue?
https://github.com/SURFnet/rd-sram-integration/issues/197
we talked there and agreed on this
the short answer (just because of the master/slave database situation)

**Comment by thepeak99** [comment]
Wait I thought we agreed to put a small cache in the GroupBackend class, at the end you ended up wrapping the GroupManager with the famous proxy object :D 

I mean it's okay too, I just want to know why didn't you just put it in the GroupBackend as we discussed instead of creating a proxy object.

**Comment by soltanireza65** [comment]
> 

I just started to do as we agreed, but then I just realized that the  manager is listening to the `postDelete` event and modifying `cachedGroups`
and also I just decided not to use GroupBackend directly inside the controller (controller should not directly deal with DB)

**Comment by thepeak99** [comment]
Perfecto! So could you please delete the commented code (or comment why it should stay there :D) and we can merge :grimacing:

**Review by thepeak99 — APPROVED** [review]

**Pull request merged**